infra/app/envs/prod/main.tf

```diff
@@ -24,7 +24,7 @@ terraform {
   required_providers {
     aws = {
       source  = "hashicorp/aws"
-      version = ">=4.20.1"
+      version = ">=4.59.0"
```

Comment:
It would be nice if this wasn't spread out across so many files. This is a problem. Add an @TODO here. We could try to restructure as 1 terraform state bucket for the project. We might need to create subdirs in the s3 bucket. To think about it. Esp to think about how it intersects with terraform workspaces